### PR TITLE
Feature: Hibernate on property annotations

### PR DESCRIPTION
**Pull request by cardil**
A showcase of using Hibernate with clean architecture with clean separation from domain logic. This implementation utilize MapStruct to map domain model to persistence model.

### CODE REVIEW

**Comment by cardil** [comment]
Please approve @coi-gov-pl/java-team

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        } catch (Throwable e) {
+            System.out.println("- Error downloading");
+            e.printStackTrace();
+            System.exit(1);
+        }
+    }
+
+    private static void downloadFileFromURL(String urlString, File destination) throws Exception {
+        URL website = new URL(urlString);
+        ReadableByteChannel rbc;
```
Comment (darek8686) [inline .mvn/wrapper/MavenWrapperDownloader.java:102]: You can replace this with try-with-resources
``try (ReadableByteChannel rbc = Channels.newChannel(website.openStream()); FileOutputStream fos = new FileOutputStream(destination);) {
...
}``

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        System.out.println("- Using base directory: " + baseDirectory.getAbsolutePath());
+
+        // If the maven-wrapper.properties exists, read it and check if it contains a custom
+        // wrapperUrl parameter.
+        File mavenWrapperPropertyFile = new File(baseDirectory, MAVEN_WRAPPER_PROPERTIES_PATH);
+        String url = DEFAULT_DOWNLOAD_URL;
+        if(mavenWrapperPropertyFile.exists()) {
+            FileInputStream mavenWrapperPropertyFileInputStream = null;
+            try {
+                mavenWrapperPropertyFileInputStream = new FileInputStream(mavenWrapperPropertyFile);
```
Comment (darek8686) [inline .mvn/wrapper/MavenWrapperDownloader.java:63]: You can replace this with try-with-resources
`try (FileInputStream mavenWrapperPropertyFileInputStream = new FileInputStream(mavenWrapperPropertyFile)) {...}`

Comment (cardil) [reply .mvn/wrapper/MavenWrapperDownloader.java:63]: Yup. It's generated code - Maven Wrapper :-)

Thank you!

**Pull request merged**